### PR TITLE
Sass task to compile sass files utilising either the sass command line tool or the scssphp compiler package.

### PR DESCRIPTION
**Pull request by kenguest**
Compile sass files utilising either the sass command line tool or the scssphp compiler package.

Supporting documentation and unit tests are provided.

This work is based on pull requests #206  and #151  by @JostBaron  and @pstuart2 respectively. It also has other significant enhancements such as short-hand attributes for setting styles etc.

### CODE REVIEW

**Comment by mrook** [comment]
It looks like this PR shares some commits with #501 - wrong branch?

**Comment by mrook** [comment]
Same comment as in #501 applies here I'm afraid.

**Comment by kenguest** [comment]
I think that's taken care of now...

File: classes/phing/tasks/ext/SassTask.php [hunk 1]
```diff
@@ -1047,14 +1047,26 @@ public function loadScssphp()
...
     }
 
+    public function getNewCompiler()
+    {
+        $compiler = '\\Leafo\\ScssPhp\\Compiler';
+        $obj = call_user_func_array(
+            array($compiler, '__construct'),
+            array()
+        );
+        var_dump ($obj);
```
Comment (mrook) [inline classes/phing/tasks/ext/SassTask.php]: One var_dump remaining ;-)

Comment (kenguest) [inline classes/phing/tasks/ext/SassTask.php]: Removed now. :D

**Comment by kenguest** [comment]
@mrook  - thanks for merging that in! :-)

File: classes/phing/tasks/ext/SassTask.php [hunk 2]
```diff
@@ -1047,14 +1047,26 @@ public function loadScssphp()
...
     /**
      * Initialise and return an instance of the ScssPhp Compiler.
      *
      * @return Leafo\ScssPhp\Compiler
      */
     public function initialiseScssphp()
     {
-        $scss = new Leafo\ScssPhp\Compiler();
+        //$scss = new Leafo\ScssPhp\Compiler();
+        $scss = $this->getNewCompiler();
```
Comment (pfrenssen) [inline classes/phing/tasks/ext/SassTask.php]: This change breaks PHP7 compatibility. I'm getting the following error:

> call_user_func_array() expects parameter 1 to be a valid callback, non-static method Leafo\ScssPhp\Compiler::__construct() cannot be called statically

Comment (kenguest) [inline classes/phing/tasks/ext/SassTask.php]: Bother. I'll see what I can do about this shortly. Ultimately if it's down to ScssPhp not being PHP 7 compatible a bug fix would be better going up-stream to that project.

**Pull request merged**